### ISSUE TITLE
Error while building 

### ISSUE THREAD

**Issue by DaddyWesker** [leo-drive/proto_to_rosbag2#2]
Hello and thanks for your code.

I've tried to build your proto_to_rosbag but i've met problem when launching colcon build

```
Starting >>> rosidl_typesupport_introspection_cpp
--- stderr: pluginlib                                                                                                                              
In file included from /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros/pluginlib/pluginlib/include/pluginlib/class_loader.hpp:334,
                 from /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros/pluginlib/pluginlib/test/utest.cpp:33:
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros/pluginlib/pluginlib/include/pluginlib/./class_loader_imp.hpp: In member function ‘std::string pluginlib::ClassLoader<T>::getPackageFromPluginXMLFilePath(const string&)’:
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros/pluginlib/pluginlib/include/pluginlib/./class_loader_imp.hpp:589:39: error: no match for ‘operator==’ (operand types are ‘rcpputils::fs::path’ and ‘rcpputils::fs::path’)
  589 |     if (parent.string().empty() || (p == parent)) {
      |                                     ~ ^~ ~~~~~~
      |                                     |    |
      |                                     |    rcpputils::fs::path
      |                                     rcpputils::fs::path
In file included from /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros/pluginlib/pluginlib/include/pluginlib/class_loader.hpp:334,
                 from /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros/pluginlib/pluginlib/test/unique_ptr_test.cpp:32:
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros/pluginlib/pluginlib/include/pluginlib/./class_loader_imp.hpp: In member function ‘std::string pluginlib::ClassLoader<T>::getPackageFromPluginXMLFilePath(const string&)’:
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros/pluginlib/pluginlib/include/pluginlib/./class_loader_imp.hpp:589:39: error: no match for ‘operator==’ (operand types are ‘rcpputils::fs::path’ and ‘rcpputils::fs::path’)
  589 |     if (parent.string().empty() || (p == parent)) {
      |                                     ~ ^~ ~~~~~~
      |                                     |    |
      |                                     |    rcpputils::fs::path
      |                                     rcpputils::fs::path
make[2]: *** [CMakeFiles/pluginlib_utest.dir/build.make:76: CMakeFiles/pluginlib_utest.dir/test/utest.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:242: CMakeFiles/pluginlib_utest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/pluginlib_unique_ptr_test.dir/build.make:76: CMakeFiles/pluginlib_unique_ptr_test.dir/test/unique_ptr_test.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:214: CMakeFiles/pluginlib_unique_ptr_test.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

```
Any solution?

**Comment by xmfcx** [comment]
@DaddyWesker 

When I've created this package, the `galactic` wasn't released yet so I had to include `rolling` in it.

Now we have it, you can install `ros2 galactic` (it's possible to install it side by side with `foxy` if you need that)

And follow the new instructions: https://github.com/leo-drive/proto_to_rosbag2#run-instructions

**TLDR:**

You can simply source `galactic`, compile and run this package.

**Comment by DaddyWesker** [comment]
@xmfcx 

So it is built-in in galactic right now?

**Comment by xmfcx** [comment]
@DaddyWesker 

I've meant, you can compile it by sourcing galactic. You can simply follow the updated readme of this repository.

**Comment by DaddyWesker** [comment]
I've installed galactic instead of foxy and have tried to run fresh colcon build. Here is what i've got

```
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp: In instantiation of ‘void rclpy::ClockEvent::wait_until(std::shared_ptr<rclpy::Clock>, rcl_time_point_t) [with ClockType = std::chrono::_V2::system_clock; rcl_time_point_t = rcl_time_point_t]’:
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp:67:55:   required from here
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp:39:41: error: invalid conversion from ‘const rcl_time_point_t*’ to ‘rcl_time_point_t*’ [-fpermissive]
   39 |   rcl_ret_t ret = rcl_difference_times(&rcl_entry, &until, &delta_t);
      |                                         ^~~~~~~~~
      |                                         |
      |                                         const rcl_time_point_t*
In file included from /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp:18:
/opt/ros/galactic/include/rcl/time.h:476:22: note:   initializing argument 1 of ‘rcl_ret_t rcl_difference_times(rcl_time_point_t*, rcl_time_point_t*, rcl_duration_t*)’
  476 |   rcl_time_point_t * start, rcl_time_point_t * finish, rcl_duration_t * delta);
      |   ~~~~~~~~~~~~~~~~~~~^~~~~
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp: In instantiation of ‘void rclpy::ClockEvent::wait_until(std::shared_ptr<rclpy::Clock>, rcl_time_point_t) [with ClockType = std::chrono::_V2::steady_clock; rcl_time_point_t = rcl_time_point_t]’:
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp:101:52:   required from here
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp:39:41: error: invalid conversion from ‘const rcl_time_point_t*’ to ‘rcl_time_point_t*’ [-fpermissive]
   39 |   rcl_ret_t ret = rcl_difference_times(&rcl_entry, &until, &delta_t);
      |                                         ^~~~~~~~~
      |                                         |
      |                                         const rcl_time_point_t*
In file included from /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp:18:
/opt/ros/galactic/include/rcl/time.h:476:22: note:   initializing argument 1 of ‘rcl_ret_t rcl_difference_times(rcl_time_point_t*, rcl_time_point_t*, rcl_duration_t*)’
  476 |   rcl_time_point_t * start, rcl_time_point_t * finish, rcl_duration_t * delta);
      |   ~~~~~~~~~~~~~~~~~~~^~~~~
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/_rclpy_pybind11.cpp: In function ‘void pybind11_init__rclpy_pybind11(pybind11::module&)’:
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/_rclpy_pybind11.cpp:77:25: error: ‘RMW_QOS_DEADLINE_BEST_AVAILABLE’ was not declared in this scope; did you mean ‘RMW_QOS_DEADLINE_DEFAULT’?
   77 |     rmw_time_total_nsec(RMW_QOS_DEADLINE_BEST_AVAILABLE));
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         RMW_QOS_DEADLINE_DEFAULT
/home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/_rclpy_pybind11.cpp:79:25: error: ‘RMW_QOS_LIVELINESS_LEASE_DURATION_BEST_AVAILABLE’ was not declared in this scope; did you mean ‘RMW_QOS_LIVELINESS_LEASE_DURATION_DEFAULT’?
   79 |     rmw_time_total_nsec(RMW_QOS_LIVELINESS_LEASE_DURATION_BEST_AVAILABLE));
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         RMW_QOS_LIVELINESS_LEASE_DURATION_DEFAULT
In file included from /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.hpp:22,
                 from /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ros2/rclpy/rclpy/src/rclpy/clock_event.cpp:25:
/usr/include/c++/9/condition_variable:133:7: error: ‘bool std::condition_variable::wait_until(std::unique_lock<std::mutex>&, const std::chrono::time_point<_Clock, _Duration1>&, _Predicate) [with _Clock = std::chrono::_V2::system_clock; _Duration = std::chrono::duration<long int, std::ratio<1, 1000000000> >; _Predicate = rclpy::ClockEvent::wait_until(std::shared_ptr<rclpy::Clock>, rcl_time_point_t) [with ClockType = std::chrono::_V2::system_clock; rcl_time_point_t = rcl_time_point_t]::<lambda()>]’, declared using local type ‘rclpy::ClockEvent::wait_until(std::shared_ptr<rclpy::Clock>, rcl_time_point_t) [with ClockType = std::chrono::_V2::system_clock; rcl_time_point_t = rcl_time_point_t]::<lambda()>’, is used but never defined [-fpermissive]
  133 |       wait_until(unique_lock<mutex>& __lock,
      |       ^~~~~~~~~~
/usr/include/c++/9/condition_variable:133:7: error: ‘bool std::condition_variable::wait_until(std::unique_lock<std::mutex>&, const std::chrono::time_point<_Clock, _Duration1>&, _Predicate) [with _Clock = std::chrono::_V2::steady_clock; _Duration = std::chrono::duration<long int, std::ratio<1, 1000000000> >; _Predicate = rclpy::ClockEvent::wait_until(std::shared_ptr<rclpy::Clock>, rcl_time_point_t) [with ClockType = std::chrono::_V2::steady_clock; rcl_time_point_t = rcl_time_point_t]::<lambda()>]’, declared using local type ‘rclpy::ClockEvent::wait_until(std::shared_ptr<rclpy::Clock>, rcl_time_point_t) [with ClockType = std::chrono::_V2::steady_clock; rcl_time_point_t = rcl_time_point_t]::<lambda()>’, is used but never defined [-fpermissive]

```

**Comment by xmfcx** [comment]
Maybe previous build artifacts are left. Can you remove the build, install folders and retry?

Also make sure you source only galactic.

I've confirmed on multiple machines that this package compiles fine.

**Comment by DaddyWesker** [comment]
I've deleted those folders again, i've rechecked that i'm sourcing only galactic, i've relaunched colcon build and i've met the same error. 

Maybe i need to source some rmw implementation?

**Comment by xmfcx** [comment]
I see that you still have the ros2 folder in the /home/daddywesker/3rdPrtyLibs/rosbag_converter_proto_ws/src/ directory. Please remove that. If you see the current readme, it shouldn't be there.

**Comment by DaddyWesker** [comment]
Hm. Okay. Now it was built. That is weird. Thanks.
